pinot-common/src/main/java/com/linkedin/pinot/common/utils/retry/AttemptsExceededException.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.common.utils.retry;
+
+/**
+ * The <code>AttemptsExceededException</code> indicates that the operation does not succeed within maximum number of
+ * attempts.
+ */
+public class AttemptsExceededException extends Exception {
```

Comment:
Should probably have a parent exception class like `AttempedOperationFailureException` or something similar.